cosmos/x/evm/types/tx.go

```diff
@@ -132,3 +133,12 @@ func (etr *EthTransactionRequest) ValidateBasic() error {
 
 	return nil
 }
+
+// GetAsEthTx is a helper function to get an EthTx from a sdk.Tx.
+func GetAsEthTx(tx sdk.Tx) *coretypes.Transaction {
```

Comment:
Just use the one on the transaction tbh.

Reply:
How else do you do it? The sdk.Tx only has `GetMsgs()` on it. One of te messages is EthTransactionRequest

Reply:
oh I see what you did yeah makes sense.